app/views/posts/index.html.erb

```diff
@@ -12,14 +12,29 @@
   <tbody>
     <% @posts.each do |post| %>
       <tr>
+      <td><h1><%= post.title %></h1>
+
+      <p><%= post.body %></p>
+      <% if user_signed_in?%>
+      <p>Written by <%= post.author%></p>
+      <% end %></td>
         <td><%= link_to 'Show', post %></td>
         <td><%= link_to 'Edit', edit_post_path(post) %></td>
         <td><%= link_to 'Destroy', post, method: :delete, data: { confirm: 'Are you sure?' } %></td>
       </tr>
+      
+      
+      
     <% end %>
   </tbody>
 </table>
 
 <br>
 
 <%= link_to 'New Post', new_post_path %>
+<% if user_signed_in?%>
+<%= link_to 'Sign Out', destroy_user_session_path, method: :delete %>
+<% else %>
+<%= link_to 'Sign In', new_user_session_path %>
+<%= link_to 'New User', new_user_registration_path %>
+<% end%>
```

Comment:
**Optional**

- [ ] It will also be nice to move the highlighted logic to a helper. The reason is given above.